src/gcovr/formats/gcov/workers.py

```diff
@@ -95,20 +104,20 @@ class Workers:
     add method and will run until work is complete
     """
 
-    def __init__(self, number: int, context: Callable[[], Dict[str, Any]]):
+    def __init__(self, number: int, context: Callable[[], Dict[str, Any]]) -> None:
         if number < 1:
             raise AssertionError("At least one executer is needed.")
-        self.q: Queue = Queue()
+        self.q: "Queue[QueueContent]" = Queue()
         self.lock = RLock()
         self.exceptions: List[str] = []
         self.contexts = [context() for _ in range(0, number)]
-        self.workers = [
+        self.workers: List[Thread] = [
             Thread(target=worker, args=(self.q, c, self)) for c in self.contexts
         ]
         for w in self.workers:
             w.start()
```

Comment:
Checked the docs and would give this a try. Also using a return value instead of using the arguments for the result. Maybe we can switch to https://docs.python.org/3/library/concurrent.futures.html#processpoolexecutor which sidesteps the GIL but the directory lock needs to be changed to work between processes.